apis/network/v1beta1/internetgateway_types.go

```diff
@@ -43,24 +51,28 @@ func (v *VPCIDReferencerForInternetGateway) Assign(res resource.CanReference, va
 		return errors.New(errResourceIsNotInternetGateway)
 	}
 
-	ig.Spec.VPCID = value
+	ig.Spec.ForProvider.VPCID = value
 	return nil
 }
 
 // InternetGatewayParameters define the desired state of an AWS VPC Internet
 // Gateway.
 type InternetGatewayParameters struct {
 	// VPCID is the ID of the VPC.
-	VPCID string `json:"vpcId,omitempty"`
+	VPCID string `json:"vpcId"`
```

Comment:
I don't think this will work if only `vpcIdRef` is supplied. This API server will reject creation if `vpcId` is not supplied.